contracts/integrations/test/simulation/simulation.ts

```diff
@@ -0,0 +1,38 @@
+import { BlockchainBalanceStore } from '@0x/contracts-exchange';
+import { GlobalStakeByStatus, StakeStatus, StakingPoolById, StoredBalance } from '@0x/contracts-staking';
+import * as _ from 'lodash';
+
+import { DeploymentManager } from '../utils/deployment_manager';
+import { AssertionResult } from '../utils/function_assertions';
+
+// tslint:disable:max-classes-per-file
+
+export class SimulationEnvironment {
```

Comment:
Thought: This makes a lot of sense as an environment for a staking system. This said, I think that `SimulationEnvironment` should be more abstract than this is. Maybe we can make this the `StakingSimulationEnvironment`? For example, when we extend the framework to function for the exchange contracts, we may not always want these properties.

Reply:
There isn't really any overhead for the unused properties and it's already a pretty lightweight class, so I'm going to leave as is for now

Reply:
Okay, that's fine. I do think it should be made more abstract in a future PR though.